Changes to SDPA to support no kv cache export

tarun292 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #7532
* #7531
* __->__ #7530

Differential Revision: [D67878163](https://our.internmc.facebook.com/intern/diff/D67878163)

tarun292 committed:
Changes to SDPA to support no kv cache export

[ghstack-poisoned]

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878163).

dvorjackz commented on examples/models/llama/source_transformation/sdpa.py:
```diff
@@ -54,9 +58,20 @@ def forward(
+                q,
+                k,
+                v,
+                input_pos,
```
Tbh I think you can just set this to 0, should work for the no kv cache text decoder as well since it represents start position, so you don't need to set input_pos to torch.tensor(0) in your other pr

tarun292 committed:
Update on "Changes to SDPA to support no kv cache export"


Differential Revision: [D67878163](https://our.internmc.facebook.com/intern/diff/D67878163)

[ghstack-poisoned]

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878163).

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878163).
